websites/R/Rumble/presence.ts

```diff
@@ -0,0 +1,154 @@
+		presenceData: PresenceData = {
+			largeImageKey: Assets.Rumble,
+		};
+
+	if (
+		document.querySelector<HTMLMetaElement>("meta[property='og:url']") &&
+		document
+			.querySelector<HTMLMetaElement>("meta[property='og:url']")
+			.content.startsWith("https://rumble.com/v") &&
+		!pathname.startsWith("/videos")
+	) {
+		if (document.querySelector(".chat--header")) {
+			(presenceData.smallImageKey = Assets.Live),
+				(presenceData.smallImageText = "Live");
+		}
+
+		if (privacy) {
+			if (presenceData.smallImageKey)
+				presenceData.details = "Watching a livestream";
+			else presenceData.details = "Watching a video";
+
+			presence.setActivity(presenceData);
+			return;
+		}
+
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "Watch",
+					url: document.querySelector<HTMLMetaElement>(
+						"meta[property='og:url']"
+					).content,
+				},
+			];
+			if (presenceData.smallImageKey) presenceData.buttons[0].label += " Live";
+			else presenceData.buttons[0].label += " Video";
+		}
+
+		if (
+			channelPic &&
+			window.getComputedStyle(document.querySelector("i")).backgroundImage !==
+				"none"
+		) {
+			presenceData.largeImageKey = window
+				.getComputedStyle(document.querySelector("i"))
+				.backgroundImage.slice(
+					5,
+					window.getComputedStyle(document.querySelector("i")).backgroundImage
+						.length - 2
+				);
+		}
+
+		presenceData.details =
+			document.querySelector<HTMLVideoElement>(".h1").textContent;
+		presenceData.state = document.querySelector<HTMLVideoElement>(
+			".media-heading-name"
+		).textContent;
+	} else if (
+		document.querySelector<HTMLMetaElement>("meta[property='og:url']") &&
+		document
+			.querySelector<HTMLMetaElement>("meta[property='og:url']")
+			.content.startsWith("https://rumble.com/c/")
+	) {
+		presenceData.details = "Viewing channel";
+
+		if (privacy) {
+			presence.setActivity(presenceData);
+			return;
+		}
+
+		presenceData.state = document.querySelector("h1").textContent;
+
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "View Channel",
+					url: document.querySelector<HTMLMetaElement>(
+						"meta[property='og:url']"
+					).content,
+				},
+			];
+		}
+		if (channelPic && document.querySelector(".channel-header--thumb")) {
+			presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+				".channel-header--thumb"
+			).currentSrc;
+		}
+	} else if (
+		document.querySelector<HTMLMetaElement>("meta[property='og:url']") &&
+		document
+			.querySelector<HTMLMetaElement>("meta[property='og:url']")
+			.content.startsWith("https://rumble.com/user/")
+	) {
+		presenceData.details = "Viewing user";
+
+		if (privacy) {
+			presence.setActivity(presenceData);
+			return;
+		}
+
+		presenceData.state = document.querySelector("h1").textContent;
+
+		if (buttons) {
+			presenceData.buttons = [
+				{
+					label: "View User",
+					url: document.querySelector<HTMLMetaElement>(
+						"meta[property='og:url']"
+					).content,
```

Comment:
Since this query is used many times in this presence, put it in a variable near the start for more concise code.